pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -52,61 +52,26 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &kubevirt.KubeVirt{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
-	}
-
-	err = c.Watch(&source.Kind{Type: &cdi.CDI{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
-	}
-
-	err = c.Watch(&source.Kind{Type: &networkaddons.NetworkAddonsConfig{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
-	}
-
-	// SSP needs to handle few types; SSP components are intentionally split in few CRs
-	err = c.Watch(&source.Kind{Type: &sspv1.KubevirtCommonTemplatesBundle{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
-	}
-	err = c.Watch(&source.Kind{Type: &sspv1.KubevirtNodeLabellerBundle{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
-	}
-	err = c.Watch(&source.Kind{Type: &sspv1.KubevirtTemplateValidator{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
-	}
-	err = c.Watch(&source.Kind{Type: &kwebuis.KWebUI{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &hcov1alpha1.HyperConverged{},
-	})
-	if err != nil {
-		return err
+	// Watch secondary resources
+	for _, resource := range []runtime.Object{
+		&kubevirt.KubeVirt{},
+		&cdi.CDI{},
+		&networkaddons.NetworkAddonsConfig{},
+		&sspv1.KubevirtCommonTemplatesBundle{},
+		&sspv1.KubevirtNodeLabellerBundle{},
+		&sspv1.KubevirtTemplateValidator{},
+		&kwebuis.KWebUI{},
+	} {
+		err = c.Watch(&source.Kind{Type: resource}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &hcov1alpha1.HyperConverged{},
+		})
+		if err != nil {
+			return err
+		}
 	}
 
-	return nil
+	return err
```

Comment:
I think your logic is right in that `err` will always be `nil` here, but I think it reads better as `nil`.  WDYT?